### DISCUSSION TITLE
Get Install referrals for both ios and android in react native firebase

### DISCUSSION THREAD

**Discussion by pruthvirajkarurcodewave** [invertase/react-native-firebase#4948] [Q&A]
How to get campaign referrals for both ios and android in react native firebase.

To generate a campaign link we have to use 
For android:
https://developers.google.com/analytics/devguides/collection/android/v4/campaigns#google-play-url-builder 
For iOS
https://developers.google.com/analytics/devguides/collection/ios/v3/campaigns#url-builder

For android you just have to have the package name and other campaign related details.
But for ios we need to add the Google Analytics Property ID:  which should be in "UA-XXXX-Y" this format. Where as for firebase projects we will have just the 9 digit Google Analytics Property ID something like "123456789".
If we give that property id to the ios campaign url generator it throws an error saying invalid format asks to give in  "UA-XXXX-Y" this format.

For android i did generate the campaign url from the above link and the results are not shown in firebase but are present in google analytics.

To summarise,  i have these questions. one on how to generate campaign url for ios. and how to capture it for ios.
Other is why is my campaign source not captured in firebase but in google analytics automatically?

**Comment by mikehardy** [comment]
Hi there! Sorry I don't know personally

I believe this is best asked on stack overflow, with appropriate firebase tags, for these reasons: it is a general firebase question not react-native-firebase specific, firebase specific questions are directed (by google firebase team, if I understand correctly) to stackoverflow where they actually participate frequently, and the community there is active, whereas you are not likely to receive answers

I would try that and I bet you get some response
